examples/restore_login.py

```diff
@@ -0,0 +1,77 @@
+import asyncio
+import json
+import os
+import sys
+
+from nio import AsyncClient, LoginResponse
+
+CONFIG_FILE = "restore_login_config.json"
+
+# Check out main() below to see how it's done.
+
+def write_details_to_disk(resp: LoginResponse) -> None:
+    """Writes the required login details to disk so we can log in later without
+    using a password.
+
+    Arguments:
+        resp {LoginResponse} -- the successful client login response.
+    """
+    # open the config file in write-mode
+    with open(CONFIG_FILE, "w") as f:
+        # write the login details to disk
+        json.dump(
+            {
+            "access_token": resp.access_token,
+            "device_id": resp.device_id,
+            "user_id": resp.user_id
+            },
+            f
+        )
+
+
+async def send_example(client: AsyncClient, room_id: str) -> None:
```

Comment:
Perhaps subclassing the `AsyncClient` and making this a method of our class is the more Pythonic way to do this.

Reply:
My concern about using stuff like subclassing is that it suggests a certain design pattern: "you should use Classes and Hierarchies and Subclass Our Structures!"

Not to say the pattern is wrong, but only that I want to make the examples... easy? Small? I think I'd prefer to not subclass it so that users are dealing with a kind of MVP example that they can easily fix and use how they'd like, but I could also do that by moving that method straight into `main()`, which is probably better